crates/sovereign-sdk/adapters/risc0/src/guest/mod.rs

```diff
@@ -26,30 +26,50 @@ pub use zkvm::Risc0Guest;
 
 // This is a dummy impl because T: ZkvmGuest where T: Zkvm.
 impl Zkvm for Risc0Guest {
-    type CodeCommitment = Risc0MethodId;
+    type CodeCommitment = Vec<u8>;
 
     type Error = anyhow::Error;
 
     fn verify(
-        serialized_proof: &[u8],
+        journal: &[u8],
         code_commitment: &Self::CodeCommitment,
     ) -> Result<Vec<u8>, Self::Error> {
-        let receipt: Receipt = bincode::deserialize(serialized_proof)?;
-        env::verify(code_commitment.0, receipt.journal.bytes.as_slice())
-            .expect("Guest side verification error should be Infallible");
-        Ok(receipt.journal.bytes)
+        let cc = vec_to_u32_array(code_commitment.clone()).unwrap();
+        env::verify(cc, journal).expect("Guest side verification error should be Infallible");
+        Ok(journal.to_vec())
     }
 
     fn verify_and_extract_output<Da: sov_rollup_interface::da::DaSpec, Root: BorshDeserialize>(
         serialized_proof: &[u8],
         code_commitment: &Self::CodeCommitment,
     ) -> Result<sov_rollup_interface::zk::StateTransition<Da, Root>, Self::Error> {
-        let receipt: Receipt = bincode::deserialize(serialized_proof)?;
-        env::verify(code_commitment.0, receipt.journal.bytes.as_slice())
-            .expect("Guest side verification error should be Infallible");
+        // let receipt: Receipt = bincode::deserialize(serialized_proof)?;
+        // env::verify(code_commitment.0, receipt.journal.bytes.as_slice())
+        //     .expect("Guest side verification error should be Infallible");
+
+        // Ok(BorshDeserialize::deserialize(
+        //     &mut receipt.journal.bytes.as_slice(),
+        // )?)
+        unimplemented!()
```

Comment:
write the reason in unimplemented!()

"Risc0 guests should already have access to the journal before verifying proof" or smth like that

Reply:
also remove comment